src/dmd/mtype.d

```diff
@@ -5033,6 +5035,24 @@ extern (C++) final class TypeFunction : TypeNext
                     return stc;
                 }
             }
+
+            // Check escaping through `this`
+            if (tthis && tthis.isMutable())
+            {
+                auto tb = tthis.toBasetype();
+                AggregateDeclaration ad;
+                if (tb.ty == Tclass)
+                    ad = (cast(TypeClass)tb).sym;
+                else if (tb.ty == Tstruct)
+                    ad = (cast(TypeStruct)tb).sym;
+                else
+                    assert(0);
+                foreach (VarDeclaration v; ad.fields)
+                {
+                    if (v.hasPointers())
+                        return stc;
```

Comment:
This looks suspicious.

Reply:
don't know what you mean

Reply:
![image](https://user-images.githubusercontent.com/4370550/37257882-49b9711c-256f-11e8-8acd-69308529d822.png)

Reply:
What about it? Please don't make me guess.

Reply:
Oh my point was only that these two lines are uncovered, thus `return stc` is never called and thus all code that you added has never an effect for the testcase.

Reply:
never mind, I see

Reply:
Should be covered now.